client_app/src/components/shelter/RequestForHelp.js

```diff
@@ -38,65 +39,65 @@ export default function RequestForHelp() {
   }, [])
 
   const handleSaveEvent = () => {
-    if (newEvent.volunteersRequired && newEvent.start && newEvent.end) {
-      if (parseInt(newEvent.volunteersRequired) <= 0) {
-        setVolunteerError("'Number of Volunteers Required' needs to be greater than 0");
-        return;
-      }
+    if (!newEvent.volunteersRequired || !newEvent.start || !newEvent.end){
+      setVolunteerError("Required fields cannot be empty.");
+      return;
+    }
 
-      const isOverlapping = myEvents.some((event) => {
-        if (
-          selectedEvent &&
-          event.start === selectedEvent.start &&
-          event.end === selectedEvent.end
-        ) {
-          return false;
-        }
-    
-        const eventStart = new Date(event.start);
-        const eventEnd = new Date(event.end);
-        return newEvent.start < eventEnd && newEvent.end > eventStart;
-      });
-    
-      if (isOverlapping) {
-        setVolunteerError("This shift overlaps with an existing shift.");
-        return;
+    if (parseInt(newEvent.volunteersRequired) <= 0){
+      setVolunteerError("Number of Volunteers required needs to be greater than 0.");
+      return;
+    }
+
+    const isOverlapping = myEvents.some((event) => {
+      if(selectedEvent && event.start == selectedEvent.start && event.end == selectedEvent.end){
+        return false;
       }
 
-      const volunteersRegex = /\s\(Volunteers:\s\d+\)$/;
-      const volunteersText = ` (Volunteers: ${newEvent.volunteersRequired})`;
-      const formattedEvent = {
-        ...newEvent,
-        title: selectedEvent
-          ? newEvent.title.replace(volunteersRegex, "") + volunteersText 
-          : newEvent.title + volunteersText,
-      };
+      const eventStart = new Date(event.start);
+      const eventEnd = new Date(event.end);
+      return newEvent.start < eventEnd && newEvent.end > eventStart;
+    }); 
 
-      if (selectedEvent) {
-        setEvents((prev) =>
-          prev.map((ev) =>
-            ev.start === selectedEvent.start && ev.end === selectedEvent.end
-              ? {
-                  ...formattedEvent,
-                  title: formattedEvent.title,
-                }
-              : ev
-          )
-        );
```

Comment:
When I create a new event thorough the UI, I see that it gets sent to the server and event details are printed to the console on the server. However, the event does not show up in the calendar view of UI, like it used to. I think that this might be the code that made the newly created event show up.